Prolangle/Services/SnippetRevealer.cs

```diff
@@ -0,0 +1,46 @@
+using System.Text.RegularExpressions;
+
+namespace Prolangle.Services;
+
+/// <param name="guessGame">The underlying game. Also used as seed source for jitter.</param>
+/// <param name="sourceCode">The source code to conceal and subsequently reveal.</param>
+/// <param name="useJitter">Whether to introduce random shift, rather than always reveal code from the same center.</param>
+public partial class SnippetRevealer(GuessGame guessGame, string sourceCode, bool useJitter)
+{
+	private double RevealedProgress { get; set; } = 0.0;
+
+	public string RevealMore()
+	{
+		RevealedProgress += 0.1;
+
+		// replace all non-whitespaces with a bullet
+		var concealedCode = NonWhitespaceRegex().Replace(sourceCode, "•");
+
+		// reveal RevealedProgress % from the middle of the snippet using the original source code
+		var revealStart = (int)(concealedCode.Length / 2f - concealedCode.Length * RevealedProgress / 2);
+		var revealEnd = (int)(concealedCode.Length / 2f + concealedCode.Length * RevealedProgress / 2);
+
+		if (useJitter)
+		{
+			// randomly move the center of revealed code around a little, to make snippets harder to remember
+			var random = new Random(guessGame.Seeder());
```

Comment:
Shouldn't the `SnippetRevealer` get its own seeder? I think this is a shadow dependency and might cause a state change in `guessGame` without that being the intent

Reply:
Yeah, e4a1baba952f745ecffd374ff83fb034708bb354 is probably nicer.

Reply:
(Thoughts?)